controller/src/beerocks/master/db/db.cpp

```diff
@@ -4227,3 +4227,28 @@ bool db::update_client_entry_in_persistent_db(
 
     return true;
 }
+
+bool db::add_client_entry_and_update_counter(
```

Comment:
well actually in this specific case it is very important - since I wish to distinguish that this private function manages the `m_persistent_db_clients_count` unlike and any API calling it must not update the counter by itself.

Reply:
This is tightly related to my BLOB claim:
Once you have such a huge class with so many unrelated entities, tasks and concerns - you simply start to add signs, signals, hints to make sure other parts of the same thing are acting correctly. This is a bad sign.
The opposite would be that you have a small class, managing a small number of inter-related entities, and then this question, this concern, wouldn't rise at all. It is clear that among the 5-10 functions one is responsible to this member.

Reply:
This is true - and as mentioned in my other reply - there is a separate task for refactoring the db class for that purpose.